libs/transform.js

```diff
@@ -17,30 +18,33 @@ class TransformBase {
   /**
    * @param {string} species What this transform is called.
    * @param {string[]} requires What datasets are required before this can run?
+   * @param {string} produces What dataset does this transform produce?
    * @param {Boolean} input Does this transform require input?
-   * @param {Boolean} output Does this transform produce output?
+   * @param {Boolean} output Does this transform produce input?
    */
-  constructor (species, requires, input, output) {
+  constructor(species, requires, produces, input, output) {
     util.check(species && (typeof species === 'string') &&
-               Array.isArray(requires) &&
-               requires.every(x => (typeof x === 'string')),
-               `Bad parameters to constructor`)
+      Array.isArray(requires) &&
+      requires.every(x => (typeof x === 'string')) &&
+      ((produces === null) || (typeof produces === 'string')),
+      `Bad parameters to constructor`)
     this.species = species
     this.requires = requires
+    this.produces = produces
     this.input = input
     this.output = output
   }
 
-  equal (other) {
+  equal(other) {
     return (other instanceof TransformBase) &&
       (this.species === other.species)
   }
 
-  equalColumns (other) {
+  equalColumns(other) {
     util.check('columns' in this,
-               `This object must have columns`)
+      `This object must have columns`)
```

Comment:
Please stick to the original indentation unless we decided to make a global change.